tensorflow/python/framework/tensor_util.py

```diff
@@ -520,7 +521,17 @@ def make_tensor_proto(values, dtype=None, shape=None, verify_shape=False,
     if nparray.size * nparray.itemsize >= (1 << 31):
       raise ValueError(
           "Cannot create a tensor proto whose content is larger than 2GB.")
-    tensor_proto.tensor_content = nparray.tobytes()
+    
+    #ensure sys and dtype endianness match
+    sys_is_le = sys.byteorder == 'little'
+    numpy_is_le = numpy_dtype.byteorder == '<'
```

Comment:
It looks like byteorder can also be "=" or '|'. 
[https://numpy.org/doc/stable/reference/generated/numpy.dtype.byteorder.html]

For '=' (native byte order) and '|' (NA -- e.g., 'i1' or 'S2' dtypes), we don't need to do any conversion.